core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -267,17 +269,19 @@ class ReplicaFetcherThread(name: String,
     }
   }
 
+  def withinQuota(partition: TopicAndPartition): Boolean = {
+    !(quota.isThrottled(partition) && quota.isQuotaExceeded)
+  }
+
   protected def buildFetchRequest(partitionMap: Map[TopicAndPartition, PartitionFetchState]): FetchRequest = {
     val requestMap = mutable.Map.empty[TopicPartition, JFetchRequest.PartitionData]
-
-    partitionMap.foreach { case ((TopicAndPartition(topic, partition), partitionFetchState)) =>
-      if (partitionFetchState.isActive)
-        requestMap(new TopicPartition(topic, partition)) = new JFetchRequest.PartitionData(partitionFetchState.offset, fetchSize)
+    partitionMap.foreach { case ((partition, partitionFetchState)) =>
+      if (partitionFetchState.isActive && withinQuota(partition))
+        requestMap(new TopicPartition(partition.topic, partition.partition)) = new JFetchRequest.PartitionData(partitionFetchState.offset, fetchSize)
     }
-
+    trace("Created a fetch request for partitions " + partitionMap.keys.map(_.toString))
```

Comment:
There is already a trace statement in AbstractFetcherThread that logs each fetch request. Do we still need this trace logging?